far/vmenu.cpp

```diff
@@ -2129,253 +2331,139 @@ void VMenu::ShowMenu(bool IsParent)
 
-					if (ItemWidth + 2 > m_Where.width())
-						ItemWidth = m_Where.width() - 2;
+		if (!PrevItem && !NextItem)
+			break;
 
-					GotoXY(m_Where.left + (m_Where.width() - 2 - ItemWidth) / 2, Y);
-					Text(concat(L' ', fit_to_left(Items[I].Name, ItemWidth), L' '));
-				}
-			}
+		if (PrevItem == BoxSymbols[BS_V1])
+		{
+			if (NextItem == BoxSymbols[BS_V1])
+				separator[I + (BoxType == NO_BOX?1:2) + 1] = BoxSymbols[BS_C_H1V1];
 			else
-			{
-				if (m_BoxType!=NO_BOX)
-				{
-					SetColor(Colors[VMenuColorBox]);
-					Text(BoxChar);
-					GotoXY(m_Where.right, Y);
-					Text(BoxChar);
-				}
-
-				GotoXY(m_Where.left + (m_BoxType == NO_BOX? 0 : 1), Y);
-
-				FarColor CurColor;
-				if ((Items[I].Flags&LIF_SELECTED))
-					CurColor = Colors[Items[I].Flags & LIF_GRAYED? VMenuColorSelGrayed : VMenuColorSelected];
-				else
-					CurColor = Colors[Items[I].Flags & LIF_DISABLE? VMenuColorDisabled : (Items[I].Flags & LIF_GRAYED? VMenuColorGrayed : VMenuColorText)];
-
-				SetColor(CurColor);
-
-				string strMenuLine;
-				wchar_t CheckMark = L' ';
-
-				if (Items[I].Flags & LIF_CHECKED)
-				{
-					if (!(Items[I].Flags & 0x0000FFFF))
-						CheckMark = 0x221A;
-					else
-						CheckMark = static_cast<wchar_t>(Items[I].Flags & 0x0000FFFF);
-				}
-
-				strMenuLine.push_back(CheckMark);
-				strMenuLine.push_back(L' '); // left scroller (<<) placeholder
-				const auto PrefixSize = strMenuLine.size();
-
-				size_t HotkeyVisualPos = string::npos;
-				auto MenuItemForDisplay = CheckFlags(VMENU_SHOWAMPERSAND)? Items[I].Name : HiText2Str(Items[I].Name, &HotkeyVisualPos);
+				separator[I + (BoxType == NO_BOX?1:2) + 1] = BoxSymbols[BS_B_H1V1];
+		}
+		else if (NextItem == BoxSymbols[BS_V1])
+		{
+			separator[I + (BoxType == NO_BOX?1:2) + 1] = BoxSymbols[BS_T_H1V1];
+		}
+	}
+}
 
-				MenuItemForDisplay.erase(0, Items[I].ShowPos);
+void VMenu::ApplySeparatorName(const MenuItemEx& CurItem, string& separator) const
+{
+	if (CurItem.Name.empty() || separator.size() <= 3)
+		return;
 
-				if (HotkeyVisualPos != string::npos)
-				{
-					if (HotkeyVisualPos < Items[I].ShowPos)
-						HotkeyVisualPos = string::npos;
-					else
-						HotkeyVisualPos -= Items[I].ShowPos;
-				}
+	auto NameWidth{ std::min(CurItem.Name.size(), separator.size() - 2) };
+	auto NamePos{ (separator.size() - NameWidth) / 2 };
 
-				// fit menu string into available space
-				bool ShowRightScroller = false;
-				if (MenuItemForDisplay.size() > MaxLineWidth)
-				{
-					MenuItemForDisplay.resize(MaxLineWidth);
-					ShowRightScroller = true;
-				}
+	separator.at(NamePos - 1) = L' ';
+	separator.replace(NamePos, NameWidth, fit_to_left(CurItem.Name, NameWidth));
+	separator.at(NamePos + NameWidth) = L' ';
+}
 
-				strMenuLine += MenuItemForDisplay;
+void VMenu::DrawRegularItem(const MenuItemEx& CurItem, const item_layout& Layout, const int Y, std::vector<int>& HighlightMarkup, const string& BlankLine) const
```

Comment:
Any reason BlankLine can't be const string_view?

Reply:
No reasons. Fixed. Thank you. Omitted `const` because `string_view` is effectively const anyway. Let me know if you agree.

Reply:
const is meaningful even on string_view: it means that you can't (accidentally) change the view itself, e.g. by calling remove_suffix or reassigning it altogether.

Reply:
OK, I should've known your tastes better. 🙂